react/__tests__/pickInstallments.test.ts

```diff
@@ -40,3 +43,68 @@ describe('pickInstallments', () => {
     expect(pickedInstallments[index].NumberOfInstallments).toBe(15)
   })
 })
+
+describe('pickMaxInstallmentsOption', () => {
+  it('should pick the installments plan with the highest NumberOfInstallments', () => {
+    const pickedInstallmentsPlan = pickMaxInstallmentsOption(
+      installmentsListMastercardMax
+    )
+
+    expect(pickedInstallmentsPlan.NumberOfInstallments).toBe(15)
+    expect(pickedInstallmentsPlan.PaymentSystemName).toBe('Mastercard')
+  })
+
+  it('should pick the installments plan with the highest NumberOfInstallments, from a certain payment system', () => {
+    const pickedInstallmentsPlan = pickMaxInstallmentsOption(installmentsList, {
+      paymentSystemName: 'Customer Credit',
+    })
+
+    expect(pickedInstallmentsPlan.NumberOfInstallments).toBe(3)
+    expect(pickedInstallmentsPlan.PaymentSystemName).toBe('Customer Credit')
+  })
+
+  it('should pick an installments plan with a certain number of installments', () => {
+    const pickedInstallmentsPlan = pickMaxInstallmentsOption(
+      installmentsListMastercardMax,
+      {
+        installmentsQuantity: 14,
+      }
+    )
+
+    expect(pickedInstallmentsPlan.NumberOfInstallments).toBe(14)
+  })
+})
+
+describe('pickMaxInstallmentsOptionWithNoInterest', () => {
+  it('should pick the installments plan with the highest NumberOfInstallments and no interest', () => {
+    const pickedInstallmentsPlan = pickMaxInstallmentsOptionWithNoInterest(
+      installmentsListMastercardMax
+    )
+
+    expect(pickedInstallmentsPlan.NumberOfInstallments).toBe(14)
+    expect(pickedInstallmentsPlan.PaymentSystemName).toBe('Visa')
```

Comment:
Seems something weird, `installmentsListMastercardMax` with payment system `Visa`. Is this right?

Reply:
This is actually right! 😅  It's easier to understand with the IDE open.
This `installmentsListMastercardMax` mock does indeed have an option with `PaymentSystemName` set to `Mastercard`, and it is the max one! But since this test is calling `pickMaxInstallmentsOptionWithNoInterest`, and the `Mastercard` option has interest, we shouldn't select it, but the `Visa` one instead.